src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/FlashModelBase.hpp

```diff
@@ -101,6 +112,31 @@ class FlashModelBase
    */
   virtual void checkTablesParameters( real64 pressure, real64 temperature ) const = 0;
 
+  /**
+   * @brief Print the table(s) in the log and/or CSV files when requested by the user.
+   * @param tableData The target table to be printed
+   * @param pvtOutputOpts Struct containing output options
+   */
+  void outputPVTTableData( TableFunction const * tableData, TableOutputOptions const pvtOutputOpts )
+  {
+    if( pvtOutputOpts.writeInLog &&  tableData->numDimensions() <= 2 )
+    {
+      TableTextFormatter textFormatter;
+      GEOS_LOG_RANK_0( textFormatter.toString( *tableData ));
+    }
+    if( pvtOutputOpts.writeCSV || ( pvtOutputOpts.writeInLog && tableData->numDimensions() >= 3 ) )
+    {
+      string const filename = tableData->getName();
+      std::ofstream logStream( joinPath( OutputBase::getOutputDirectory(), filename + ".csv" ) );
+      GEOS_LOG_RANK_0( GEOS_FMT( "CSV Generated to inputFiles/compositionalMultiphaseWell/{}/{}.csv \n",
+                                 OutputBase::getOutputDirectory(),
+                                 filename ));
+
+      TableCSVFormatter csvFormatter;
+      logStream << csvFormatter.toString( *tableData );
+    }
+  }
```

Comment:
There is a lot of duplication with `PVTFunctionBase`, that is why this mistake occured.
Should / can we plan for a refactor for these 2 files to remove these duplications? Inheritance seems like an obvious option but may not be the best.

Reply:
This will happen in a future PR